dht/i_partitioner.hh

```diff
@@ -847,8 +847,7 @@ namespace std {
 template<>
 struct hash<dht::token> {
     size_t operator()(const dht::token& t) const {
-        const auto& b = t._data;
-        return read_le<size_t>(reinterpret_cast<const char*>(b.data()));
+        return t._data;
```

Comment:
This changes the return value of the function, if we run on a big-endian machine. We shouldn't depend on the hash value being stable across nodes, but how can we verify this?

Reply:
I don't understand. previously b.data() was in le and read_le was converting it to host byte ordering. Now t._data is in host byte ordering. Isn't it the same?

Reply:
Actually the data was in big endian order. So we would reverse the byte order unconditionally.

Now you're unconditionally not reversing it.

To be safe, let's unconditionally reverse it now. Something like read_le(read_be()) should work.

Reply:
Hmm. read_le calls le_to_cpu which in turn calls le64toh this will do nothing on little-endian processors and reverse bytes on big-endian processors, no? This means that on little-endian returned value will have bytes in big-endian order and on big-endian processors it will have bytes in little-endian order. Or do I miss something?

Little-endian CPU case:
b.data() -> bytes in network (big-endian) order
read_le calls le64toh which does nothing -> returning int64_t composed of bytes in unchanged order = network (big-endian) order

Big-endian CPU case: 
b.data() -> bytes in network (big-endian) order
read_le calls le64toh which reverses bytes -> returning int64_t composed of bytes in reversed order = little-endian order

This means you are right and we always reverse the order :) - I will use read_le(read_be()). Thanks.